test/observer_test.dart

```diff
@@ -88,94 +23,128 @@ void main() async {
+  test('Observe multiple entities', () async {
+    Completer completer;
+    var expectedEvents = 0;
+    var typesUpdates = <Type, int>{}; // number of events per entity type
 
-    box.putMany(simpleStringItems);
+    final stream = env.store.subscribeAll();
 
-    box.remove(1);
+    final subscription = stream.listen((entityType) {
+      print('Entity updated: $entityType');
+      expectedEvents--;
 
-    // update value
-    final entity2 = box.get(2);
-    entity2.tString = 'Dva';
-    box.put(entity2);
+      if (typesUpdates[entityType] == null) {
+        typesUpdates[entityType] = 0;
+      }
+      typesUpdates[entityType]++;
 
-    final box2 = Box<TestEntity2>(store);
-    box2.put(TestEntity2());
-    box2.remove(1);
-    box2.put(TestEntity2());
-
-    expect(callbackAnyTypeCounter, 6);
-    bindings.obx_observer_close(observer);
-  });
-
-  test('Observe single entity', () async {
-    final callback =
-        Pointer.fromFunction<obx_observer_single_type>(callbackSingleType);
-    final observer = bindings.obx_observe_single_type(
-        store.ptr, testEntityId, callback, randomPtr);
+      if (expectedEvents == 0) {
+        completer.complete();
+      }
+    });
 
+    // expect two events after one put() and one putMany()
+    expectedEvents = 3;
+    completer = Completer();
+    box.put(simpleStringItems[0]);
+    Box<TestEntity2>(env.store).put(TestEntity2());
     box.putMany(simpleStringItems);
-    simpleStringItems.forEach((i) => box.put(i));
-    simpleNumberItems.forEach((i) => box.put(i));
-
-    expect(callbackSingleTypeCounter, 13);
-    bindings.obx_observer_close(observer);
+    await completer.future.timeout(defaultTimeout);
+    expect(expectedEvents, 0);
+    expect(typesUpdates.keys, unorderedEquals([TestEntity, TestEntity2]));
+    expect(typesUpdates[TestEntity], 2);
+    expect(typesUpdates[TestEntity2], 1);
+
+    // cancel the subscription
+    await subscription.cancel();
+
+    // make sure there are no more events after cancelling
+    expectedEvents = 1;
+    completer = Completer();
+    box.put(simpleStringItems[0]);
+    expect(completer.future.timeout(defaultTimeout),
+        throwsA(isA<TimeoutException>()));
+    expect(expectedEvents, 1); // note: unchanged, no events received anymore
   });
 
-  tearDown(() {
-    env.close();
+  test('Observer pause/resume', () async {
+    final testPauseResume = (Stream stream) async {
+      Completer completer;
+      final subscription = stream.listen((_) {
+        completer.complete();
+      });
+
+      // triggers when listening
+      completer = Completer();
+      box.put(simpleStringItems[0]);
+      await completer.future.timeout(defaultTimeout);
+
+      // won't trigger when paused
+      subscription.pause();
+      completer = Completer();
+      box.put(simpleStringItems[0]);
+      expect(completer.future.timeout(defaultTimeout),
+          throwsA(isA<TimeoutException>()));
+
+      // triggers when resumed
```

Comment:
Maybe note that it only triggers once, not twice (no buffered event as the underlying Observable should stop events while stream is paused).